apps/arweave/src/ar_block.erl

```diff
@@ -86,6 +87,24 @@ generate_size_tagged_list_from_txs(TXs) ->
 		)
 	).
 
+generate_size_tagged_tx_ids(TXs) ->
+	lists:reverse(
+		element(
+			2,
+			lists:foldl(
+				fun(TX = #tx{id = TXID}, {Pos, List}) ->
+						End = Pos + get_tx_data_size(TX),
```

Comment:
You can just do `#tx.data_size` like it's done in `generate_size_tagged_list_from_txs`.